internal/common/function/function.go

```diff
@@ -135,6 +135,10 @@ func ValidateBackupSpec(ctx context.Context, clientInstance client.Client, oadpN
 		}
 	}
 
+	if nonAdminBackup.Spec.BackupSpec.VolumeSnapshotLocations != nil {
```

Comment:
Should there be also a check if `&& len(nonAdminBackup.Spec.BackupSpec.VolumeSnapshotLocations) > 0` ?

Reply:
I think current implementation does not allow non admin user to set `spec.backupSpec.volumeSnapshotLocations`

with your idea, it would allow to set to an empty list (but velero coudl still propably change the value on backup side)

which is better?

Reply:
Ok your is better as it needs to be nil.

One last bit - will this be still nil if the sync controller recreates it from s3 Backup object spec?

Reply:
no, only special case is BSL ref that is updated https://github.com/migtools/oadp-non-admin/blob/2d5e4e7ae5ce34063aa3d9e49c399cc8a8818870/internal/controller/nonadminbackupsynchronizer_controller.go#L148-L153